### PR TITLE
Requirements review

### LINKED ISSUE #1107: Review requirements and decide which should be updated
At the very least we should update Django to the latest build. Even though we don't use `DEBUG` in production we can't prevent any of our users from doing so.

https://requires.io/github/mirumee/saleor/requirements/?branch=master

### PR DESCRIPTION
**Pull request by salwator**
I've reviewed requirements and updated most of them, including security django fix.
Closes #1107

### CODE REVIEW

**Comment by elwoodxblues** [comment]
@salwator Please use [Github keywords](https://help.github.com/articles/closing-issues-using-keywords/) for closing issues, when a PR resolves a particular issue. Also please follow [this](https://github.com/erlang/otp/wiki/Writing-good-commit-messages) guidelines for commit messages.

File: requirements.in [hunk 1]
```diff
@@ -9,7 +9,7 @@ django-cache-url>=1.0.0
 django-countries>=4.1
 django-mptt>=0.7.1
 django-payments>=0.11.0,<0.12a0
-django-prices>=0.4.11,<0.5a0
+django-prices>=0.4.11
```
Comment (patrys) [inline requirements.in]: Should we drop the other "lower than" guards for consistency?

**Comment by patrys** [comment]
What about the ancient GeoIP database?

File: requirements.in (hunk 1 above)
Comment (salwator) [reply requirements.in]: Didn't want to remove guards I wasn't sure about. I will do it if there is no reason for specific api freeze anywhere.
What about Pillow library, is there any reason behind 4.0.0 upper bound?

Comment (patrys) [reply requirements.in]: The `<0.x` guards predate `pip-compile`. I think now that we do explicit version bumps it does not make sense to have them unless they are followed with a comment explaining the reason (ie. "# see issue #0000").

**Pull request merged**